### PR TITLE
std::net: adding acceptfilter feature for netbsd/freebsd.

### PR DESCRIPTION
**Pull request by devnexen**
similar to linux's ext deferaccept, to filter incoming connections before accept.

### CODE REVIEW

**Comment by Amanieu** [comment]
@devnexen Can you fix the CI failure?

File: library/std/src/sys/pal/unix/net.rs [hunk 1]
```diff
@@ -453,6 +453,37 @@ impl Socket {
         Ok(raw as u32)
     }
 
+    #[cfg(any(target_os = "freebsd", target_os = "netbsd"))]
+    pub fn set_acceptfilter(&self, name: &CStr) -> io::Result<()> {
+        if !name.to_bytes().is_empty() {
+            const AF_NAME_MAX: usize = 16;
+            let mut buf = [0; AF_NAME_MAX];
+            for (src, dst) in name.to_bytes().iter().zip(&mut buf[..AF_NAME_MAX - 1]) {
+                *dst = *src as i8;
```
Comment (taiki-e) [inline library/std/src/sys/pal/unix/net.rs:462]: This should be `as c_char`. (c_char is u8 by default on most non-Apple/non-Windows ARM/AArch64/PowerPC/RISC-V/s390x/Hexagon targets)

```console
# with empty crate
$ cargo check -Z build-std --target aarch64-unknown-freebsd
error[E0308]: mismatched types
   --> /Users/taiki/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/sys/pal/unix/net.rs:465:27
    |
461 |             for (src, dst) in name.to_bytes().iter().zip(&mut buf[..AF_NAME_MAX - 1]) {
    |                                                               --- here the type of `buf` is inferred to be `[i8; 16]`
...
465 |             arg.af_name = buf;
    |             -----------   ^^^ expected `[u8; 16]`, found `[i8; 16]`
    |             |
    |             expected due to the type of this binding
    |
    = note: expected array `[u8; 16]`
               found array `[i8; 16]`
```

Comment (taiki-e) [reply library/std/src/sys/pal/unix/net.rs:462]: I opened https://github.com/rust-lang/rust/pull/122983 to fix this.

**Pull request merged**